client/src/pages/SearchMovie/index.js

```diff
@@ -1,42 +1,73 @@
-import React, { useState } from 'react'
+import React, { useState, useCallback, useMemo } from 'react'
 import debounce from 'lodash.debounce'
 
-import { SearchMovieContainer } from './styled'
+import { SearchMovieContainer, ClearImg, Button } from './styled'
 import { BebasText } from '../../styles/Text'
 import { SearchBar, SearchInput, SearchImg } from '../../styles/SearchBar'
 import { useConfiguration } from '../../providers/Configuration'
 import { authFetch } from '../../auth'
 import Results from './Results'
+import { useMovies } from '../../providers/Movies'
 
 const SearchMovie = () => {
   const { apiUrl } = useConfiguration()
+  const { movies } = useMovies()
+  const [inputValue, setInputValue] = useState('')
   const [movieResults, setMovieResult] = useState([])
+  const [loadSearchMovies, setLoadSearchMovie] = useState(false)
 
-  const debouncedSearch = debounce(
-    (currentValue) => searchMovies(currentValue),
-    1000
-  )
+  const searchMovies = async (currentValue, movieIds) => {
+    console.log('callback')
 
-  async function searchMovies(currentValue) {
+    setLoadSearchMovie(true)
     if (!currentValue) {
       setMovieResult([])
+      setLoadSearchMovie(false)
       return
     }
+
     try {
+      console.log(currentValue)
       const response = await authFetch(
         `${apiUrl}/movies?search=${currentValue}`
       )
       const data = await response.json()
+
+      console.log('hello')
       if (data.Search) {
+        for (const movieData of data.Search) {
```

Comment:
ideally, this logic should be in some sort of utility function that returns a new array.
```
const movieResultsWithStatus = addWatchlistStatus(data.Search)
setMovieResults(movieResultsWithStatus)
```